internal/formatter/json/driver.go

```diff
@@ -0,0 +1,41 @@
+package json
+
+import (
+	"encoding/json"
+
+	"github.com/docker/app/internal/formatter"
+	"github.com/docker/app/internal/yaml"
+	composeloader "github.com/docker/cli/cli/compose/loader"
+	composetypes "github.com/docker/cli/cli/compose/types"
+	"github.com/pkg/errors"
+)
+
+func init() {
+	formatter.Register("json", &Driver{})
+}
+
+// Driver is the json implementation of formatter drivers.
+type Driver struct{}
+
+// Apply creates a JSON document from the source config.
+func (d *Driver) Apply(config *composetypes.Config) (string, error) {
+	// The config type has field name and omitempty annotations for YAML,
```

Comment:
Should we put them there (upstream) and just do a `json.Marshal` then ?

Reply:
@shin- not really. We should consider the `compose` part of  the `docker/cli` as its own lib… (I'm lobbying to get this code out some day but that's another story).  And act as "is this something that "library" should do/allow ? And I feel, marshalling to json is something we could allow in there. (cc @thaJeztah)

There is already stuff "only for" `docker/app` in there (like extracting variables from a composefile).

Reply:
I've been attempting this today, and it's just as painful as I expected it to be. I'm pretty much stuck because the `time.Duration`-typed fields will only marshal as numbers, but the Compose schema only allows strings for these same fields. I've tried adding a wrapper around time.Duration like suggested here: https://stackoverflow.com/a/48051946, but that breaks `mapstructure` in turn.

Overall that's confirmed my intuition that adding a layer of JSON annotations in there is the wrong approach, but let me know if you wanna try and take a crack at it and I can share what I have.